libraries/AP_HAL_ChibiOS/hwdef/CrazyF405/README.md

```diff
@@ -0,0 +1,73 @@
+# CrazyF405HD ELRS 1-2SAIO Flight Controller
+
+The CrazyF405HD ELRS 1-2S AIO is a flight controller produced by [Happymodel](https://www.happymodel.cn/index.php/2023/05/26/crazyf405hd-elrs-1-2s-aio-fc-built-in-uart-elrs-receiver-and-12a-blheli_s-esc/.)
+
+## Features
+
+ - MCU: STM32F405RGT6, 168MHz
+ - Gyro: BMI270 (SPI)
+ - 1Mb Onboard Flash
+ - BEC output: 5V, 2A@4V
+ - Barometer: BMP280
+ - OSD: 
+ - 3 UARTS: (UART1, UART2 ,UART6)
+ - 5 PWM outputs (4 motor outputs used internally for integrated 4-in-1 ESC and 1 integrated LED)
+ - Integrated 4-in-1 BlueJay ESC
+ - Firmware target: BetaflightF4
+
+## Pinout
+
+![CrazyF405HD ELRS 1-2S AIO Board](CrazyF405HD_pinout.jpg "CrazyF405HD ELRS 1-2S AIO")
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+|Name|Pin|Function|
+|:-|:-|:-|
+|SERIAL0|COMPUTER|USB|
+|SERIAL1|RX1/TX1|UART1 ( DMA-enabled)|
+|SERIAL2|TX2/RX2|UART2 (ELRS, internal)
+|SERIAL6|RX6|UART6 (DMA-disabled)|
+
+## RC Input
+
+RC input is configured on the on-board ELRS on UART2 or through (UART2_RX/UART2_TX) pins. It supports all serial RC protocols.
+
+## OSD Support
+The CrazyF405HDAIO is optimized for Digital HD FPV and does not require the analog OSD chip (MAX7456); OSD data is transmitted via MSP to the digital VTX.
+
+## PWM Output
+
+The Carzyf405HD AIO has 4 PWM outputs internally connected to its 4-in-1 ESC. The pads for motor output are M1 to M4 on the board. All 4 outputs support bi-directional DShot and DShot, as well as all PWM types. The default configuration is for DShot using the already installed BlueJay firmware.
```

Comment:
your hwdef does NOT support bi-directional DShot. it may, but does not now

Reply:
Ya ,so i will mention about the DSHOT only